src/posix/platform/shared_lib.cpp

```diff
@@ -0,0 +1,272 @@
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ * @brief
+ *   This file implements the processing of OpenThread stack in thread safe mode.
+ */
+
+#include "openthread-posix-config.h"
+
+#include <openthread/config.h>
+
+#include <errno.h>
+#include <pthread.h>
+#include <sys/stat.h>
+#include <syslog.h>
+#include <unistd.h>
+
+#include <openthread/logging.h>
+#include <openthread/tasklet.h>
+#include <openthread/thread.h>
+
+#include <common/code_utils.hpp>
+#include <lib/platform/exit_code.h>
+#include <openthread/openthread-system.h>
+
+#if OPENTHREAD_POSIX_CONFIG_MULTITHREAD_SUPPORT_ENABLE
+
+#define MULTIPLE_INSTANCE_MAX 10
+#define MAX_TTY_LEN 100
+#define MAX_RADIO_URL_LEN 200
+#define MAX_INTERFACE_LEN 100
+
+static otInstance     *gInstance = NULL;
+static pthread_t       gThreadId;
+static pthread_mutex_t gLock;
+static bool            gTerminate = 0;
+
+typedef struct Param
+{
+    char     comPort[MAX_TTY_LEN];
+    uint16_t debugLevel;
+} Param;
+
+static bool getRadioURL(char *comPort)
+{
+    struct stat st;
+    char        radioDevice[MAX_RADIO_URL_LEN] = {0};
+
+    /* For multiple devices and intefaces,
+     * we will align the index of them.
+     */
+    sprintf(radioDevice, "/dev/%s", comPort);
+    if (stat(radioDevice, &st) == 0)
+    {
+        syslog(LOG_INFO, "radio device found [%s]", radioDevice);
+        return true;
+    }
+    return false;
+}
+
+static int getInterface(void)
+{
+    struct stat st;
+    int         i                        = 0;
+    char        iface[MAX_INTERFACE_LEN] = {0};
+
+    for (i = 0; i < MULTIPLE_INSTANCE_MAX; i++)
+    {
+        sprintf(iface, "/sys/class/net/wpan%d", i);
+        if (stat(iface, &st) == 0)
+        {
+            syslog(LOG_INFO, "Interface is already used [%s]", iface);
+            continue;
+        }
+        else
+        {
+            syslog(LOG_INFO, "find empty interface [%s]", iface);
+            break;
+        }
+    }
+    if (i == MULTIPLE_INSTANCE_MAX)
+    {
+        syslog(LOG_CRIT, "Interface reached max count...Not able to create new interface");
+        return -1;
+    }
+
+    return i;
+}
+
+void *pThreadOtMainLoop(void *arg)
+{
+    pthread_mutex_lock(&gLock);
+    otPlatformConfig platformConfig;
+    int              interfaceIdx                = 0;
+    char             radioUrl[MAX_RADIO_URL_LEN] = {0};
+    char             iface[MAX_INTERFACE_LEN]    = {0};
+    Param           *initParam                   = (Param *)arg;
+
+    syslog(LOG_INFO, "Inside pThreadOtMainLoop");
+
+    interfaceIdx = getInterface();
+    sprintf(iface, "wpan%d", interfaceIdx);
+    syslog(LOG_INFO, "interface found [%s]", iface);
+
+    if (!getRadioURL(initParam->comPort))
+    {
+        syslog(LOG_CRIT, "radio device not found");
+        return NULL;
```

Comment:
Per style guide we should avoid having multiple `return` from same function/method.
Can use `code_utils.hpp` macros `VerifyOrExit()` or `ExitNow()`,

There are other functions in this file with multiple `return`s.